tests/Integration/Bridge/TolerantParser/Reflection/ReflectionPropertyTest.php

```diff
@@ -277,6 +277,46 @@ function (ReflectionPropertyCollection $properties) {
                     $this->assertEquals(Type::fromString('Bar\Foo'), $properties->get('bar')->inferredTypes()->best());
                 },
             ],
+            'Understands root namespace' => [
+                <<<'EOT'
+<?php
+
+class Foobar
+{
+    private $property1;
+
+    public function __construct()
+    {
+        $this->property1 = new \DateTime();
+    }
+}
+EOT
+                ,
+                'Foobar',
+                function ($properties) {
+                    $this->assertEquals('\DateTime', $properties->get('property1')->inferredTypes()->best()->short());
```

Comment:
not sure this is correct, the short name should never have a leading slash?

Reply:
Mmm interesting. So the short name should always remove the trailing slash and then it would be in the extension responsible of the return type/parameter type hint to add the trailing slash or import the use statement?

Good for me :+1: